src/coreclr/nativeaot/System.Private.Reflection.Execution/src/Internal/Reflection/Execution/ExecutionEnvironmentImplementation.ManifestResources.cs

```diff
@@ -82,17 +82,12 @@ private unsafe Stream ReadResourceFromBlob(ResourceInfo resourceInfo)
                 throw new BadImageFormatException();
             }
 
-            checked
+            if (resourceInfo.Index < 0 || resourceInfo.Length < 0 || (uint)(resourceInfo.Index + resourceInfo.Length) > cbBlob)
```

Comment:
Since you're on this - could you make this an assert? This info comes from the executable image we're currently running - it's not user provided. (Same for the if check above this.)

Reply:
Converted to an assert but kept the previous `if` unchanged.